cli/pkg/autodetect/kind.go

```diff
@@ -37,34 +38,48 @@ type DetectionArguments struct {
 	KubeClient    *kube.Client
 }
 
-type Detector interface {
-	Detect(ctx context.Context, args DetectionArguments) (Kind, error)
+type ClusterKindDetector interface {
+	Detect(ctx context.Context, args DetectionArguments) bool
+	Kind() Kind
 }
 
-func KubernetesClusterProduct(ctx context.Context, kc string, client *kube.Client) (Kind, string) {
+type ClusterDetails struct {
+	Kind       Kind
+	K8SVersion *version.Version
+}
+
+var (
+	ErrCannotDetectK8sVersion  = fmt.Errorf("cannot detect k8s version")
+	ErrCannotDetectClusterKind = fmt.Errorf("cannot detect cluster kind")
+)
+
+func DetectK8SClusterDetails(ctx context.Context, kc string, client *kube.Client) (ClusterDetails, error) {
 	details := k8sutils.GetCurrentClusterDetails(kc)
 	serverVersion, err := client.Discovery().ServerVersion()
-	kubeVersion := fmt.Sprintf("%s.%s", serverVersion.Major, serverVersion.Minor)
-	gitServerVersion := ""
-	if err == nil {
-		gitServerVersion = serverVersion.GitVersion
+	if err != nil {
+		return ClusterDetails{}, errors.Join(ErrCannotDetectK8sVersion, err)
```

Comment:
we use this later on: `errors.Is(err, autodetect.ErrCannotDetectClusterKind)`. will it work if the rr is `errors.Join`?